docs/central-upgrade.rst

```diff
@@ -76,14 +77,179 @@ You'll be asked to confirm the removal of all dangling images. Agree by typing t
+  head files/service/config.json.template
+
+If you see ``"host": "postgres"``, you are using the default database. If ``host`` is set to anything else, you are using a custom database server.
+
+.. tabs::
+
+  .. tab:: Default database
+
+    .. warning::
+      Before starting:
+
+      #. :doc:`Back up your server <central-backup>`.
+      #. Make sure you have some time available in case something goes wrong (we recommend at least 2 hours). You may want to announce a maintenance window.
+      #. Log into the web interface and make a quick note of some of the data you see, such as submission counts and latest submission dates. You may want to use this information to do a quick spot check after the upgrade is finished.
+      #. Review upgrade notes for all versions between your current version and the version you are upgrading to.
+      #. Read the instructions at the top of this section carefully and **make sure you are actually using the default database configuration**. Following these instructions with a custom database setup could result in perceived data loss.
+
+    #. **Log into your server.** If you used our :doc:`DigitalOcean installation steps <central-install-digital-ocean>`, we suggest reviewing the section :ref:`central-install-digital-ocean-build` as a reminder, or if you can't remember your password to start at the top of that section to reset your password.
+
+    #. **Get the latest infrastructure version.** This is a typical upgrade step.
+
+       .. code-block:: bash
+
+          cd central
+          git pull
+
+       .. note::
+
+          If you have made local changes to the files, you may have to start with ``git stash``, then run ``git stash pop`` after you perform the ``pull``. If you aren't sure, run ``git pull`` and it will tell you.
+
+    #. **Get the latest client and server.** This is also a typical upgrade step and ensures that Frontend and Backend are up to date.
+
+       .. code-block:: bash
+
+          git submodule update -i
+
+    #. **Check that you have enough disk space available.** If you are prompted for a password, enter the system superuser password (not a Central password). You will see a message about how much space is required and if you have enough free space to proceed.
+
+       .. code-block:: bash
+
+          sudo ./files/postgres14/upgrade/check-available-space
+
+       *If you don't have enough space,* **stop here** and resume when you have increased the disk space available. You may achieve this by clearing out data you don't need (e.g., logs) or by increasing the total disk space available (e.g., by :ref:`adding external storage <central-install-digital-ocean-external-storage>`).
+
+    #. **Create a file to prove that you're carefully reading these instructions.** This is required to continue.
+
+       .. code-block:: bash
+
+          touch ./files/allow-postgres14-upgrade
+
+    #. **Stop Central.** Central needs to be stopped in order to make a clean copy of the database.
```

Comment:
You should be able to build, then stop. That will minimize your downtime. That's what we've done historically, but this should be tested.

Reply:
I'm now wondering whether this is a good idea. That means things can happen between backup and stop. Maybe our standard recommendation should be to stop everything, back up, then do what needs to be done. Folks who are more advanced can do whatever they want.